lib/src/pub_license/pub_license.dart

```diff
@@ -0,0 +1,134 @@
+/// Enables checking a package's license from pub.dev.
+///
+/// This library is intented to be used by Very Good CLI to help extracting
+/// license information. The existance of this library is likely to be
+/// ephemeral. It may be obsolete once [pub.dev](https://pub.dev/) exposes
+/// stable license information in their official API; you may track the
+/// progress [here](https://github.com/dart-lang/pub-dev/issues/4717).
+library pub_license;
+
+import 'package:html/dom.dart' as html_dom;
+import 'package:html/parser.dart' as html_parser;
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+/// The pub.dev [Uri] used to retrieve the license of a package.
+Uri _pubPackageLicenseUri(String packageName) =>
+    Uri.parse('https://pub.dev/packages/$packageName/license');
+
+/// {@template pub_license_exception}
+/// An exception thrown by [PubLicense].
+/// {@endtemplate}
+class PubLicenseException implements Exception {
+  /// {@macro pub_license_exception}
+  const PubLicenseException(String message)
+      : message = '[pub_license] $message';
+
+  /// The exception message.
+  final String message;
+}
+
+/// {@template pub_license}
+/// Enables checking pub.dev's hosted packages license.
+/// {@endtemplate}
+class PubLicense {
+  /// {@macro pub_license}
+  PubLicense({
+    @visibleForTesting http.Client? client,
+    @visibleForTesting
+    html_dom.Document Function(
+      dynamic input, {
+      String? encoding,
+      bool generateSpans,
+      String? sourceUrl,
+    })? parse,
```

Comment:
Shouldn't we set a type def for this, even if it is private?

Reply:
We could but we would have to ignore the lint warning, are we happy with doing so?